HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,711 @@
+---
+hip: 
+title: Frictionless Airdrops
+author: Richard Bair <@rbair23>, Nana Essilfie-Conduah <@nana-ec>
+working-group: Jasper Potts <@jasperpotts>, Atul Mahamuni <@atul-hedera>, Michael Tinker <@tinker-michaelj>, Leemon Baird <@lbaird>, Ty Smith <@ty-swirldslabs>, Nick Poorman <@nickpoorman>, Ali Nikan <@alinik4n>, Steven Sheehy <@steven-sheehy>
+requested-by: DEXs, Wallets, External Web3 Users
+type: Standards Track
+category: Core, Service, Mirror
+needs-council-approval: Yes
+status: Draft
+created: 2024-2-25
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/905
+updated: 
+requires: 
+replaces: 655, 777
+superseded-by: 
+---
+
+## Abstract
+
+This HIP enables frictionless airdrops of both fungible and non-fungible tokens by removing the requirement to pre-associate tokens with the receiver's account, and by allowing the sender to create associations on the receiver's account automatically at transfer time by paying for any token-associations created as a result of the airdrop. The sovereignty of account holders who do not wish to receive unsolicited tokens is preserved by supporting the concept of pending token transfers to such an account. Potential token receivers may claim their token airdrop while available if desired and airdrop senders may cancel their sent airdrop should they change their mind.
+
+With this HIP, accounts will no longer prepay for unused token association slots. Association fees will be charged for those slots only when a token is actually associated with an account.
+
+This HIP also introduces the concept of token rejection. Here an undesired token class value is returned to the treasury with all custom fees waived.
```

Comment:
`token airdrop rejection`

Reply:
No just token rejection.
An account can reject any token in their balance.
Not to be confused with token cancel airdrop, whereby a sender of an airdrop can can their airdrop if it is still in pending state.

Reply:
You know, I just became curious about this.  The HTS token just magically appears back in the token type's treasury account.  Does that happen in other cases?  That is, if the back end of whatever is managing that token type is trying to keep track of what's extant can it get confused?  In the general user stories there's nothing about "[as a network user|as a treasury account owner] I'd want to be able to query the mirror node for all token rejects that landed back in my treasury account" - should there be?

Reply:
A treasury could "track" token rejects by looking at transfers in to the account. A Reject is not significantly different from any other transfer, except it can avoid certain fees otherwise paid to the treasury account by "giving back" the entire balance to that same treasury account.

Reply:
It's not magically, it's an actual transfer. So the transaction record will show the transfer of tokens from sender to treasury.

Reply:
Fair point. 
As a transaction type the mirror node will support query of instances.
Existing accounts API queries will support the query you're referring to